spec/coerce_spec.rb

```diff
@@ -180,6 +180,10 @@ class Param2 < T::Struct
       infos = T::Coerce[T::Array[ParamInfo]].new.from([{name: 'b', skill_ids: []}])
       T.assert_type!(infos, T::Array[ParamInfo])
       expect(infos.first.name).to eql 'b'
+
+      expect {
+        T::Coerce[ParamInfo2].new.from({a: nil, b: nil})
```

Comment:
why would it return ShapeError? I'd expect a ValueError, because `notes` has a default value
```
    class ParamInfo2 < T::Struct
      const :a, Integer
      const :b, Integer
      const :notes, T::Array[String], default: []
    end
```

Reply:
Hmm currently, ValueError only happens when coercion fails. Here we get an error because of the hard-sorbet-error configuration rejecting two props of the `T::Struct` a,b for being nil... I agree it makes more sense for it to be a ValueError, however we cannot really control it only using `T::Coerce::Configuration`

Reply:
How does it raise ShapeError though? ShapeError is raised manually by the code in this library when something doesn't match a Hash or an Array type.

Reply:
But based on what you said, I think we should change "ValueError" to CoerceError for clarity. If it is TypeError people know it comes from Sorbet. We only raise the errors we know of.